src/documentation/execution-model.md

```diff
@@ -66,95 +62,126 @@ Instead of processing all elements simultaneously, the elements in a
-of `ParDo1` for a given element must be processed on the same worker thread.
+**Figure 4:** Two transforms in sequence and their corresponding input collections.
+
+Figure 5 shows how these dependently parallel transforms might execute. The
+first worker executes `ParDo1` on the elements in bundle A (which results in
+bundle C), and then executes `ParDo2` on the elements in bundle C. Similarly,
+the second worker executes `ParDo1` on the elements in bundle B (which results
+in bundle D), and then executes `ParDo2` on the elements in bundle D.
 
 ![bundling_multi_gantt.svg]({{ site.baseurl }}/images/execution_model_bundling_multi_gantt.svg)
 
-For example, two `ParDo` transforms in sequence may be _dependently parallel_ if
-the runner chooses to execute the consuming transform on each output from the
-producing transform without altering the bundling in between.
+**Figure 5:** Two workers execute dependently parallel ParDo transforms.
 
 Executing transforms this way allows a runner to avoid redistributing elements
-between workers, saving on communication costs. However, the maximum parallelism
+between workers, which saves on communication costs. However, the maximum parallelism
 now depends on the maximum parallelism of the first of the dependently parallel
 steps.
 
 ### Failures within one transform
 
 If processing of an element within a bundle fails, the entire bundle fails. The
 elements in the bundle must be retried (otherwise the entire pipeline fails),
-although they need not be retried with the same bundling.
+although they do not need to be retried with the same bundling.
 
-In the following illustration, two elements (in green) were successfully
-processed. The third element’s processing failed, and there are three elements
-(in yellow) still to be processed. We see that the same elements were retried
-and processing successfully completed. Note that as shown, the retry does not
-necessarily happen in the same worker thread as the original attempt.
+For this example, we will use the `ParDo` from figure 1 that has an input
+collection with nine elements and is divided into two bundles.
 
-Because we encountered a failure while processing an element in the input
-bundle, we had to reprocess all of the elements in the input bundle. Thus, the
-entire output bundle must be thrown away since all of the results it contains
-will be recomputed.
+In figure 6, the first worker successfully processes all five elements in bundle
+A. The second worker processes the four elements in bundle B: the first two
+elements were successfully processed, the third element’s processing failed, and
+there is one element still awaiting processing.
+
+We see that the runner retries all elements in bundle B and the processing
+completes successfully the second time. Note that the retry does not necessarily
+happen on the same worker as the original processing attempt, as shown in the
+diagram.
```

Comment:
The bundling also need not remain the same. Not sure we need to call that out, but we could.

Reply:
it calls that out in the first paragraph of the section (elements must be retried, but they don't need to be retried with same bundling), so I think it's okay to leave it out here